src/dictionary.c

```diff
@@ -70,8 +220,13 @@ static void dict_init_impl(struct dictionary* d, SEXP x, bool partial) {
   } else {
     d->hash = NULL;
   }
+
+  UNPROTECT(1);
+  return d;
 }
 
+
+// Use hash from `x` but value from `d`?
 uint32_t dict_hash_with(struct dictionary* d, struct dictionary* x, R_len_t i) {
```

Comment:
Can this become static? We don't use it in other files

Reply:
I think it makes sense to export it, for instance `vec_match()` might move to another file.